fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/join/InnerJoinLAsscom.java

```diff
@@ -85,11 +91,22 @@ public Rule build() {
                 }).toRule(RuleType.LOGICAL_INNER_JOIN_LASSCOM);
     }
 
+    /**
+     * trigger rule condition
+     */
     public static boolean checkReorder(LogicalJoin<? extends Plan, GroupPlan> topJoin,
-            LogicalJoin<GroupPlan, GroupPlan> bottomJoin) {
-        return !bottomJoin.getJoinReorderContext().hasCommuteZigZag()
-                && !topJoin.getJoinReorderContext().hasLAsscom()
-                && (!bottomJoin.isMarkJoin() && !topJoin.isMarkJoin());
+            LogicalJoin<GroupPlan, GroupPlan> bottomJoin, boolean leftZigZag) {
+        if (leftZigZag) {
+            double bRows = bottomJoin.right().getGroup().getStatistics().getRowCount();
+            double cRows = topJoin.right().getGroup().getStatistics().getRowCount();
+            return bRows < cRows && !bottomJoin.getJoinReorderContext().hasCommuteZigZag()
+                    && !topJoin.getJoinReorderContext().hasLAsscom()
+                    && (!bottomJoin.isMarkJoin() && !topJoin.isMarkJoin());
```

Comment:
Add more comments to explain it. It may seem quite challenging to comprehend

Reply:
done